### PR TITLE
ref(ts): Convert organization context to ts

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationContext.tsx [hunk 1]
```diff
@@ -224,29 +250,30 @@ const OrganizationContext = createReactClass({
...
-  },
+  }
 
   getOrganizationDetailsEndpoint() {
     return `/organizations/${this.getOrganizationSlug()}/`;
-  },
+  }
 
-  getTitle() {
+  getTitle(): string {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/organizationContext.tsx:259]: I believe it's fine if you don't add a return type here 😉

Please see:

image

File: src/sentry/static/sentry/app/views/organizationContext.tsx [hunk 2]
```diff
@@ -24,57 +26,61 @@ import getRouteStringFromRoutes from 'app/utils/getRouteStringFromRoutes';
...
     };
-  },
+  }
 
   componentDidMount() {
     this.fetchData();
-  },
+  }
 
   componentDidUpdate(prevProps) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/organizationContext.tsx:85]: should we also add types here?

File: src/sentry/static/sentry/app/views/organizationContext.tsx [hunk 3]
```diff
@@ -224,29 +250,30 @@ const OrganizationContext = createReactClass({
...
+  getTitle(): string {
     if (this.state.organization) {
       return this.state.organization.name;
     }
     return 'Sentry';
-  },
+  }
 
-  renderSidebar() {
+  renderSidebar(): React.ReactNode {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/organizationContext.tsx:266]:
```suggestion
  renderSidebar() {
```

File: src/sentry/static/sentry/app/views/organizationContext.tsx [hunk 4]
```diff
@@ -102,34 +108,54 @@ const OrganizationContext = createReactClass({
...
   onProjectCreation() {
     // If a new project was created, we need to re-fetch the
     // org details endpoint, which will propagate re-rendering
     // for the entire component tree
     fetchOrganizationDetails(this.props.api, this.getOrganizationSlug(), true, true);
-  },
+  }
 
-  getOrganizationSlug() {
+  getOrganizationSlug(): string {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/organizationContext.tsx:146]:
```suggestion
  getOrganizationSlug() {
```

File: src/sentry/static/sentry/app/views/organizationContext.tsx [hunk 5]
```diff
@@ -224,29 +250,30 @@ const OrganizationContext = createReactClass({
...
 
-    return <Sidebar {...this.props} organization={this.state.organization} />;
-  },
+    const {children: _, ...props} = this.props;
+    return <Sidebar {...props} organization={this.state.organization as Organization} />;
+  }
 
   renderError() {
-    let errorComponent;
+    let errorComponent: React.ReactNode;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/organizationContext.tsx:276]:
```suggestion
    let errorComponent: React.ReactElement;
```

**Review by markstory — APPROVED** [review]
Looks good with the suggested changes.

**Pull request merged**